pkg/guacrest/openapi.yaml

```diff
@@ -99,7 +99,127 @@ paths:
           $ref: "#/components/responses/InternalServerError"
         "502":
           $ref: "#/components/responses/BadGateway"
-
+  "/v0/package/{purl}":
+    get:
+      summary: Get purls related to the specific Package URL (PURL)
+      operationId: getPackagePurlsByPurl
+      parameters:
+        - name: purl
+          in: path
+          required: true
+          description: URL-encoded Package URL (PURL)
+          schema:
+            type: string
+      responses:
+        "200":
+          description: A list of PURLs associated with the specified PURL
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: string
+                  description: Package URL (PURL)
+        "400":
+          $ref: "#/components/responses/BadRequest"
+        "500":
+          $ref: "#/components/responses/InternalServerError"
+  "/v0/package/{purl}/vulns":
+    get:
+      summary: Get vulnerabilities for a specific Package URL (PURL)
+      operationId: getPackageVulnerabilitiesByPurl
+      parameters:
+        - name: purl
+          in: path
+          required: true
+          description: URL-encoded Package URL (PURL)
+          schema:
+            type: string
+        - name: latestSBOM
+          in: query
+          required: false
+          description: Whether the query should search in the latest sbom
+          schema:
+            type: boolean
+      responses:
+        "200":
+          $ref: "#/components/responses/VulnerabilityList"
+        "400":
+          $ref: "#/components/responses/BadRequest"
+        "500":
+          $ref: "#/components/responses/InternalServerError"
+  "/v0/package/{purl}/dependencies":
+    get:
+      summary: Get dependencies for a specific Package URL (PURL)
+      operationId: getPackageDependenciesByPurl
+      parameters:
+        - name: purl
+          in: path
+          required: true
+          description: URL-encoded Package URL (PURL)
+          schema:
+            type: string
+        - name: latestSBOM
+          in: query
+          required: false
+          description: Whether the query should search in the latest sbom
+          schema:
+            type: boolean
+      responses:
+        "200":
+          $ref: "#/components/responses/PurlList"
+        "400":
+          $ref: "#/components/responses/BadRequest"
+        "500":
+          $ref: "#/components/responses/InternalServerError"
+  "/v0/artifact/{artifact}/vulns":
+    get:
+      summary: Get vulnerabilities for a specific Artifact (<algorithm>:<digest>)
+      operationId: getArtifactVulnerabilities
+      parameters:
+        - name: artifact
+          in: path
+          required: true
+          description: Artifact identifier in the format <algorithm:digest>
```

Comment:
I changed the spec, so that the endpoints don't take in the entire algorithm. Now the user just inputs the digest. 

Because of this we don't need to worry about the capitalization.